pinning_retry_worker.go

```diff
@@ -0,0 +1,36 @@
+package main
+
+import (
+	"context"
+	"time"
+
+	"github.com/application-research/estuary/util"
+)
+
+// RunPinningRetryWorker re-attempt pinning contents that have not yet been pinned after a period of time
+func (cm *ContentManager) RunPinningRetryWorker(ctx context.Context) {
+	timer := time.NewTicker(time.Hour * cm.cfg.Pinning.RetryWorker.Interval)
+	for {
+		select {
+		case <-ctx.Done():
+			return
+		case <-timer.C:
+			startContentID := 0
+			for {
+				var contents []util.Content
+				if err := cm.DB.Limit(cm.cfg.Pinning.RetryWorker.BatchLimit).Order("id ASC").Find(&contents, "pinning and not active and not failed and not aggregate and id > ? and created_at > NOW() + INTERVAL '1 HOURS'", startContentID).Error; err != nil {
```

Comment:
I'm thinking we should add a date here. We should just re-try the last 2 to 3 months of pins. We will need to run another GC routine to clean up old data.

Reply:
ok

Reply:
`and created_at > NOW() + INTERVAL '1 HOURS'` 

Why is this hard coded to `1 HOURS` ? Wouldn't it need to be dynamic based on `cm.cfg.Pinning.RetryWorker.Interval` ?

Reply:
`cm.cfg.Pinning.RetryWorker.Interval` is how often it should check,, `and created_at > NOW() + INTERVAL '1 HOURS'` is what data it should select at the interval. I will make this dynamic with a default of 2 months back